stm32/Src/phytos31.c

```diff
@@ -0,0 +1,45 @@
+/**
+ ******************************************************************************
+ * @file    sdi12.c
+ * @author  Stephen Taylor
+ * 
+ * @brief   This library is designed to read measurements from a PHYTOS-31 
+ *          sensor from METER.
+ *          https://metergroup.com/products/phytos-31/
+ * @date    4/18/2024
+ ******************************************************************************
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "phytos31.h"
+#include "transcoder.h"
+
+HAL_StatusTypeDef Phytos31Init(){
+  return ADC_init();
+}
+
+phytos_measurments Phytos31GetMeasurment(){
+  phytos_measurments measurments;
+  measurments.phytos31_raw = (double) ADC_readVoltage();
+  return measurments;
+}
+
+size_t Phytos31_measure(uint8_t *data) {
+  // get timestamp
+  time_t ts = TimestampNow();
+  phytos_measurments measurment;
+
+  // read voltage
+  measurment = Phytos31GetMeasurment();
+  double adc_voltage_float = ((double) measurment.phytos31_raw) / 1000.;
```

Comment:
What is the `/ 1000.` for? To me it seems like a calibration of sort but `Phytos31GetMeasurement` only returns raw?

Reply:
this is an error, i copied adc_measure exactly and forgot to remove the scalar.